index.js

```diff
@@ -372,60 +433,300 @@ export const sendForm = method => url => headers => form => {
+//.
+//. See also [`followRedirects`](#followRedirects), or if neither function has
+//. your desired behaviour, use [`followRedirectsWith`](#followRedirectsWith)
+//. to create a custom one.
+//.
+//. ```js
+//. retrieve ('https://example.com') ({})
+//. .pipe (chain (followRedirectsStrict (10)))
+//. ```
+export const followRedirectsStrict = followRedirectsWith ({
+  301: redirectIfGetMethod,
+  302: redirectIfGetMethod,
+  303: redirectUsingGetMethod,
+  305: redirectOriginalRequest,
+  307: redirectIfGetMethod,
+});
+
+//# followRedirects :: Number -> Response -> Future Error Response
+//.
+//. Aggressively follows redirects in violation of [RFC2616 Section 10.3][].
+//.
+//. The first argument is the maximum number of redirects that should be
+//. followed. When the maximum is reached, the response will be returned
+//. as-is, even if it might have been a followable redirect.
+//.
+//. Redirections with status codes 301, 302, and 307 are always followed
+//. without user intervention, and redirects with status code 304 are
+//. retried without conditions if the original request had any conditional
+//. headers.
+//.
+//. See also [`followRedirectsStrict`](#followRedirectsStrict), or if neither
+//. function has your desired behaviour, use
+//. [`followRedirectsWith`](#followRedirectsWith) to create a custom one.
+//.
+//. ```js
+//. retrieve ('https://example.com') ({})
+//. .pipe (chain (followRedirects (10)))
+//. ```
+export const followRedirects = followRedirectsWith ({
+  301: redirectOriginalRequest,
+  302: redirectOriginalRequest,
+  303: redirectUsingGetMethod,
+  304: retryWithoutCondition,
+  305: redirectOriginalRequest,
+  307: redirectOriginalRequest,
+});
+
+//# acceptStatus :: Number -> Response -> Future Response Response
 //.
-//. This function "tags" an [IncomingMessage][] based on a given status code.
+//. This function "tags" a [Response](#Response) based on a given status code.
 //. If the response status matches the given status code, the returned Future
 //. will resolve. If it doesn't, the returned Future will reject.
 //.
-//. The idea is that you can compose this function with one that returns an
-//. IncomingMessage, and reject any responses that don't meet the expected
-//. status code. In combination with [`responseToError`](#responseToError),
-//. you can then flatten it back into the outer Future.
-//.
-//. The usage example under the [Http](#http) section shows this.
+//. The idea is that you can compose this function with one that returns a
+//. Response, and reject any responses that don't meet the expected status
+//. code.
+
+//. In combination with [`responseToError`](#responseToError), you can then
+//. flatten it back into the outer Future. The usage example under the
+//. [Http](#http) section shows this.
 export const acceptStatus = code => res => (
-  code === res.statusCode ? resolve (res) : reject (res)
+  code === Response.message (res).statusCode ? resolve (res) : reject (res)
```

Comment:
`f (x)` formatting does not work well in cases such as this one. One option is to use redundant parens to make the grouping clear (e.g. `(f (x)).y`).